File: cogs/arrows.py
import discord
from discord.ext import commands
import globals

def setup(bot):
    bot.add_cog(Arrows(bot))


class Arrows(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @commands.group(aliases=['arw'], case_insensitive=True)
    @commands.bot_has_permissions(send_messages=True, embed_links=True)
    async def arrows(self, ctx):
        if ctx.invoked_subcommand is None:
            # create the embed
            embed = discord.Embed(
                title='Which map?',
                description='Use `val arrows [map_name]` to choose a map from the following.\n',
                color=globals.embed_color
            )
            embed.set_thumbnail(url='https://hubermjonathan-vlrnt.herokuapp.com/sova')
            embed.add_field(name='**Maps**',
                value='Bind, Haven, Split, Ascent',
                inline=False
            )

            # send the message
            await ctx.send(embed=embed)

    @arrows.command()
    @commands.bot_has_permissions(send_messages=True, embed_links=True)
    async def bind(self, ctx):
        # create the embed
        embed = discord.Embed(
            title='Sova Arrows for Bind',
            color=globals.embed_color
        )
        embed.set_thumbnail(url='https://hubermjonathan-vlrnt.herokuapp.com/sova')
        embed.add_field(name='**Coming Soon**',
            value='The bot owner is currently gathering arrow locations and will update this feature soon.',
            inline=False
        )

        # send the message
        await ctx.send(embed=embed)

    @arrows.command()
    @commands.bot_has_permissions(send_messages=True, embed_links=True)
    async def haven(self, ctx):
        # create the embed
        embed = discord.Embed(
            title='Sova Arrows for Haven',
            color=globals.embed_color
        )
        embed.set_thumbnail(url='https://hubermjonathan-vlrnt.herokuapp.com/sova')
        embed.add_field(name='**Coming Soon**',
            value='The bot owner is currently gathering arrow locations and will update this feature soon.',
            inline=False
        )

        # send the message
        await ctx.send(embed=embed)

    @arrows.command()
    @commands.bot_has_permissions(send_messages=True, embed_links=True)
    async def split(self, ctx):
        # create the embed
        embed = discord.Embed(
            title='Sova Arrows for Split',
            color=globals.embed_color
        )
        embed.set_thumbnail(url='https://hubermjonathan-vlrnt.herokuapp.com/sova')
        embed.add_field(name='**Coming Soon**',
            value='The bot owner is currently gathering arrow locations and will update this feature soon.',
            inline=False
        )

        # send the message
        await ctx.send(embed=embed)

    @arrows.command()
    @commands.bot_has_permissions(send_messages=True, embed_links=True)
    async def ascent(self, ctx):
        # create the embed
        embed = discord.Embed(
            title='Sova Arrows for Ascent',
            color=globals.embed_color
        )
        embed.set_thumbnail(url='https://hubermjonathan-vlrnt.herokuapp.com/sova')
        embed.add_field(name='**Coming Soon**',
            value='The bot owner is currently gathering arrow locations and will update this feature soon.',
            inline=False
        )

        # send the message
        await ctx.send(embed=embed)

File: src/main/java/com/hubermjonathan/vlrnt/Arrows.java
package com.hubermjonathan.vlrnt;

import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.entities.Message;
import net.dv8tion.jda.api.entities.MessageChannel;
import net.dv8tion.jda.api.entities.MessageEmbed;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;

public class Arrows extends Feature {
    public Arrows(MessageReceivedEvent event) {
        super(event);
    }

    public void sendArrowMessage(String map) {
        MessageChannel channel = getEvent().getChannel();
        Message message = getEvent().getMessage();

        if (map == null) {
            EmbedBuilder embedBuilder = new EmbedBuilder();
            embedBuilder.setTitle("Which map?");
            embedBuilder.setDescription("Use `val! arrows [map_name]` to choose a map from the following.");
            embedBuilder.addField("**Maps**", "Bind, Haven, Split, Ascent", false);
            embedBuilder.setColor(Constants.EMBED_COLOR);
            channel.sendMessage(embedBuilder.build()).queue();
            return;
        }

        switch (map.toLowerCase()) {
            case ("bind"):
                channel.sendMessage(getBindEmbed()).queue();
                break;
            case ("haven"):
                channel.sendMessage(getHavenEmbed()).queue();
                break;
            case ("split"):
                channel.sendMessage(getSplitEmbed()).queue();
                break;
            case ("ascent"):
                channel.sendMessage(getAscentEmbed()).queue();
                break;
            default:
                message.addReaction(Constants.DENY).queue();
        }
    }

    public MessageEmbed getBindEmbed() {
        EmbedBuilder embedBuilder = new EmbedBuilder();
        embedBuilder.setTitle("Sova Arrows for Bind");
        embedBuilder.setThumbnail("http://vlrnt.hubermjonathan.com/sova");
        embedBuilder.addField(
                "**Coming Soon**",
                "The bot owner is currently gathering arrow locations and will update this feature soon.",
                false
        );
        embedBuilder.setColor(Constants.EMBED_COLOR);
        return embedBuilder.build();
    }

    public MessageEmbed getHavenEmbed() {
        EmbedBuilder embedBuilder = new EmbedBuilder();
        embedBuilder.setTitle("Sova Arrows for Haven");
        embedBuilder.setThumbnail("http://vlrnt.hubermjonathan.com/sova");
        embedBuilder.addField(
                "**Coming Soon**",
                "The bot owner is currently gathering arrow locations and will update this feature soon.",
                false
        );
        embedBuilder.setColor(Constants.EMBED_COLOR);
        return embedBuilder.build();
    }

    public MessageEmbed getSplitEmbed() {
        EmbedBuilder embedBuilder = new EmbedBuilder();
        embedBuilder.setTitle("Sova Arrows for Split");
        embedBuilder.setThumbnail("http://vlrnt.hubermjonathan.com/sova");
        embedBuilder.addField(
                "**Coming Soon**",
                "The bot owner is currently gathering arrow locations and will update this feature soon.",
                false
        );
        embedBuilder.setColor(Constants.EMBED_COLOR);
        return embedBuilder.build();
    }

    public MessageEmbed getAscentEmbed() {
        EmbedBuilder embedBuilder = new EmbedBuilder();
        embedBuilder.setTitle("Sova Arrows for Ascent");
        embedBuilder.setThumbnail("http://vlrnt.hubermjonathan.com/sova");
        embedBuilder.addField(
                "**Coming Soon**",
                "The bot owner is currently gathering arrow locations and will update this feature soon.",
                false
        );
        embedBuilder.setColor(Constants.EMBED_COLOR);
        return embedBuilder.build();
    }
}
